ls1intum/Artemis
# `Programming exercises`: Fix auxiliary repositories not visible on exercise detail page

### Checklist
#### General
#### Client

### Motivation and Context
I noticed in the live server, that links to auxiliary repositories were no longer shown on the exercise detail page, even if they exist.

### Description
This was caused by overwriting the local `programmingExercise` object after a second server call. The request to `with-template-and-solution-participation` does not contain the auxiliary information anymore:

![grafik](https://user-images.githubusercontent.com/26540346/208308535-5bd99562-f7c4-4adf-bd1d-76c1a2dc1679.png)

With this PR, the second call now also sends the necessary information about auxiliary repositories to the client.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Programming Exercise with auxiliary repositories

1. Open the exercise detail page
2. Check that links to the auxiliary repositories are shown

### Review Progress
#### Code Review
#### Manual Tests

### Screenshots
![grafik](https://user-images.githubusercontent.com/26540346/208309308-149253e2-0383-42d5-b017-7d1622511014.png)
